proposal/color-4-new-spaces-js.d.ts.md

```diff
@@ -0,0 +1,540 @@
+`rgb` space.
+
+```ts
+constructor(
+  options: {
+    red: ChannelValue;
+    green: ChannelValue;
+    blue: ChannelValue;
+    alpha?: number;
+  },
+  space?: ColorSpaceRGB
+);
+```
+
+#### HSL Channel Constructor
+
+Create a new SassColor in the `hsl` color space. `space` is optional to not
+break the legacy constructor, but allowed for constructor consistency.
+
+```ts
+constructor(
+  options: {
+    hue: ChannelValue;
+    saturation: ChannelValue;
+    lightness: ChannelValue;
+    alpha?: number;
+  },
+  space?: ColorSpaceHSL
+);
+```
+
+#### HWB Channel Constructor
+
+Create a new SassColor in the `hwb` color space. `space` is optional to not
+break the legacy constructor, but allowed for constructor consistency.
+
+```ts
+constructor(
+  options: {
+    hue: ChannelValue;
+    whiteness: ChannelValue;
+    blackness: ChannelValue;
+    alpha?: number;
+  },
+  space?: ColorSpaceHWB
+);
+```
+
+#### XYZ Channel Constructor
+
+Create a new SassColor in a color space with XYZ channels -- `xyz`, `xyz-d50`,
+and `xyz-d65`. `space` is required as there is no legacy space to default to.
+
+```ts
+constructor(
+  options: {
+    x: ChannelValue;
+    y: ChannelValue;
+    z: ChannelValue;
+    alpha?: number;
+  },
+  space: ColorSpaceXYZ
+);
+```
+
+#### LCH Channel Constructor
+
+Create a new SassColor in a color space with LCH channels -- `lch` and `oklch`.
+`space` is required as there is no legacy space to default to.
+
+```ts
+constructor(
+  options: {
+    lightness: ChannelValue;
+    chroma: ChannelValue;
+    hue: ChannelValue;
+    alpha?: number;
+  },
+  space: ColorSpaceLCH
+);
+```
+
+#### LAB Channel Constructor
+
+Create a new SassColor in a color space with LAB channels -- `lab` and `oklab`.
+`space` is required as there is no legacy space to default to.
+
+```ts
+constructor(
+  options: {
+    lightness: ChannelValue;
+    a: ChannelValue;
+    b: ChannelValue;
+    alpha?: number;
+  },
+  space: ColorSpaceLAB
+);
+```
+
+```ts
+}
+```
+
+### Deprecations
+
+A number of SassColor getters only make sense for legacy colors, and so are
+being deprecated for `channel`.
+
+* `red`
+* `green`
+* `blue`
+* `hue`
+* `saturation`
+* `lightness`
+* `whiteness`
+* `blackness`
+* `alpha`
+
+In addition, `change` is deprecated in favor of `changeChannels`.
+
+## Embedded Protocol
+
+### SassColor
+
+```proto
+message SassColor {
+  // The name of a known color space.
+  string space = 1;
+
+  // The value of the first channel associated with `space`.
+  double Channel1 = 2;
+
+  // The value of the second channel associated with `space`.
+  double Channel2 = 3;
+
+  // The value of the third channel associated with `space`.
+  double Channel3 = 4;
+
+  // The color's alpha channel. Mandatory. Must be between 0 and 1,
+  // inclusive.
+  double alpha = 5;
+}
+```
+
+### Deprecations
+
+The `RgbColor`, `HslColor` and `HwbColor` SassScript values will be marked as
+deprecated in the Embedded Protocol.
```

Comment:
I'm not sure it's worth continuing to support these. We'll need to make this a breaking protocol change anyway, since the embedded compiler is going to be sending colors as `SassColor` instead of these other types, so we might as well just get rid of them entirely.